r/R/dataset-partition.R

```diff
@@ -76,7 +76,9 @@ HivePartitioning$create <- dataset___HivePartitioning
 #' calling `hive_partition()` with no arguments.
 #' @examples
 #' \donttest{
-#' hive_partition(year = int16(), month = int8())
+#' if (arrow_with_dataset()) {
```

Comment:
Dev roxygen2 has an `@examplesIf` feature that might be better here. If we put the conditional in the example code, it may suggest to the user that they also need to check this every time, and generally they shouldn't.

Reply:
I created ARROW-11849 for us to do that after this feature makes it to CRAN

Reply:
We don't need roxygen2 to be on CRAN in order to use it, the Rd it generates is valid.

Reply:
Right, roxygen2 does not run on CRAN, but I think we should avoid requiring R package contributors to use dev versions of devtools, roxygen2, etc.

Reply:
I don't feel strongly about not requiring dev roxygen, but it's fine either way.

If you don't want to use examplesIf here, I think you can leave off the `if (arrow_with_dataset())` checks everywhere since it's already inside a `\donttest`

Reply:
I initially did that, but apparently as of R 4.0.0:
>`R CMD check --as-cran` now runs `\donttest` examples

(from https://cran.r-project.org/doc/manuals/r-devel/NEWS.html)

Reply:
It does but I think CRAN doesn't do `--run-donttest` everywhere. Anyway we won't be turning off these features on CRAN except for Solaris so I think it's fine, or worth the risk so as not to pollute the examples with potentially confusing (to users) wrapping.

Reply:
Gotcha. For now, I'll remove the conditional in the example and I'll also change the `\donttest` to `\dontrun`. That way the example checks will pass on Solaris.

Reply:
Done in 6918441bba2b0ca09f2103eaf5631adc0a076a6c

Reply:
Thanks, please also remove the conditionals from the other examples. I don't feel strongly about `\donttest` vs `\dontrun` (CRAN complains about dontrun examples on initial submission, which is why they were switched to donttest in the first place, but that's not our concern now) so that's fine, but FWIW `\donttest` is safe on Solaris: [we currently "pass" Solaris checks](https://www.r-project.org/nosvn/R.check/r-patched-solaris-x86/arrow-00check.html) with the fake arrow-without-arrow build.

Reply:
Done in 26753e4e0ed08739467588cfb3ddcfe09f5aebcc. I will change the `\dontrun`s back to `\donttest` (or remove them altogether if possible) when I implement `@examplesIf`, as noted in ARROW-11849